src/test/c_api_tests.cpp

```diff
@@ -286,6 +288,151 @@ TEST_F(CAPIInference, TensorSetMovedBuffer) {
     ASSERT_EQ(tensor.setBuffer(std::move(buffer)), ovms::StatusCode::DOUBLE_BUFFER_SET);
 }
 
+TEST_F(CAPIInference, NoInputs) {
+    tensor_map_t m;
+    [[maybe_unused]] ovms::ServableMetadata sm("dummy", 1, m, m);
+    OVMS_ServableMetadata* osm = reinterpret_cast<OVMS_ServableMetadata*>(&sm);
+    uint32_t count;
+    ASSERT_EQ(sm.getVersion(), 1);
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServableMetadataGetInputCount(osm, &count));
+    ASSERT_EQ(count, 0);
+}
+
+TEST_F(CAPIInference, InferenceRequest) {
+    InferenceRequest* r = new InferenceRequest("dummy", 1);
+    size_t batchSize;
+    ASSERT_EQ(r->getBatchSize(batchSize, 1), StatusCode::INTERNAL_ERROR);
+    ASSERT_EQ(r->removeInputBuffer("dummy"), StatusCode::NONEXISTENT_TENSOR_FOR_REMOVE_BUFFER);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_InferenceRequestInputRemoveData(nullptr, "dummy"), StatusCode::NONEXISTENT_PTR);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_InferenceRequestInputRemoveData(reinterpret_cast<OVMS_InferenceRequest*>(r), nullptr), StatusCode::NONEXISTENT_PTR);
+    delete r;
+}
+TEST_F(CAPIInference, InferenceResponse) {
+    InferenceResponse* r = new InferenceResponse("dummy", 1);
+    int64_t a = 1;
+    EXPECT_THROW(r->addOutput("name", OVMS_DataType::OVMS_DATATYPE_BIN, &a, -1), std::exception);
```

Comment:
Passing -1 to addOutput cannot be validated on our side since this is overflow.
Having large value and the pointer is user's fault (passing buffer pointer and wrong buffer size).
I think we should not test it since it is not always the case that we will have exception here. This is undefined behavior so we might end up with sporadic segfaults.